### PR TITLE
Create metadata unit tests

### PR DESCRIPTION
**Pull request by murdo-moj**
* Test all the proper responses from this lambda
* Don't test the metadata validator object, as this is tested in the base image tests

### CODE REVIEW

File: containers/daap-create-metadata/src/var/task/create_metadata.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def handler(event, context):
 
     data_product_metadata = DataProductMetadata(data_product_name, logger)
 
-    if not data_product_metadata.metadata_exists:
+    if data_product_metadata.metadata_exists is False:
```
Comment (tom-webber) [inline containers/daap-create-metadata/src/var/task/create_metadata.py]:
```suggestion
    if not data_product_metadata.metadata_exists:
```
Original is more standard python (`if not .exists`), and more typically what I'd expect.

If insisting on moving to comparing with False, use ` == False`

Comment (murdo-moj) [reply containers/daap-create-metadata/src/var/task/create_metadata.py]: 😢 ok

File: containers/daap-create-metadata/tests/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,7 @@
+boto3==1.28.40
```
Comment (MatMoore) [inline containers/daap-create-metadata/tests/requirements.txt:1]: in the other lambdas I've been including the source requirements here, i.e. starting the file with

`-r ../src/var/task/requirements.txt`

and then just adding the test requirements.

I don't know if at some point we want to switch over to poetry or something else that lets us define dev dependencies vs normal dependencies in one place?

**Review by MatMoore — DISMISSED** [review]
looks good to me

**Review by MatMoore — APPROVED** [review]

**Review by MatMoore — APPROVED** [review]

**Pull request merged**